mlir/lib/Quantum/Transforms/AdjointPatterns.cpp

```diff
@@ -265,7 +265,7 @@ class AdjointGenerator {
         }
 
         for (const auto &[qubitResult, qubitOperand] :
-             llvm::zip(clone.getNonCtrlQubitResults(), gate.getNonCtrlQubitOperands())) {
+             llvm::zip(clone.getQubitResults(), gate.getQubitOperands())) {
```

Comment:
This is a very satisfying 1-line fix for what looked like a super obscure error :laughing: Thanks so much!

Reply:
Happy to help! And thanks for the bug report :)